cdap-master/src/main/java/io/cdap/cdap/master/environment/k8s/MetadataServiceMain.java

```diff
@@ -113,7 +116,11 @@ protected void addServices(Injector injector, List<? super Service> services,
       EnvironmentOptions options) {
     services.add(injector.getInstance(MetadataService.class));
     services.add(injector.getInstance(MetadataSubscriberService.class));
-    services.add(injector.getInstance(MetadataConsumerSubscriberService.class));
+    Collection<String> enabledMetadataConsumers = cConf.getStringCollection(
+        Constants.MetadataConsumer.METADATA_CONSUMER_EXTENSIONS_ENABLED_LIST);
+    if (enabledMetadataConsumers != null && !enabledMetadataConsumers.isEmpty()) {
+      services.add(injector.getInstance(MetadataConsumerSubscriberService.class));
+    }
```

Comment:
Could you please log something if we don't add this service?

Reply:
Done.